### PR TITLE
simplify yaml for averaging + summing enricher, plus test

### PR DESCRIPTION
**Pull request by ahgittin**

### CODE REVIEW

**Comment by ggrunin** [comment]
tested #605  - works well.

File: core/src/main/java/brooklyn/enricher/Enrichers.java [hunk 1]
```diff
@@ -216,13 +219,25 @@ public B fromHardcodedProducers(Iterable<? extends Entity> val) {
...
+        @SuppressWarnings({ "unchecked", "rawtypes" })
         public B computing(Function<? super Collection<S>, ? extends T> val) {
-            this.computing = checkNotNull(val);
+            this.computingSupplier = (Supplier)Suppliers.ofInstance(checkNotNull(val));
             return self();
         }
-        @SuppressWarnings({ "unchecked", "rawtypes" })
         public B computingSum() {
-            // relies of TypeCoercion of result from Number to T, and type erasure for us to get away with it!
+            this.computingSupplier = new Supplier<Function<? super Collection<S>, ? extends T>>() {
```
Comment (neykov) [inline core/src/main/java/brooklyn/enricher/Enrichers.java]: Better to use named classes to avoid rebind issues.

Comment (ahgittin) [inline core/src/main/java/brooklyn/enricher/Enrichers.java]: The `Supplier` is local to the builder, resolved before the spec is built, so I think it's fine.  The `Function` itself is a static class.

Comment (neykov) [inline core/src/main/java/brooklyn/enricher/Enrichers.java]: Oh I see it now, shouldn't trust github's enclosing scope method signature.

**Comment by neykov** [comment]
Looks good to merge.

**Pull request merged**